### PR TITLE
change meter/group selection logic and dropdown sorting

### LINKED ISSUE #600: lose selected meter when go to map
### Issue

Selecting meters that cannot be shown on the map before selecting a meter on the map graph causes them to stop being graphed.

### To reproduce

1. Select a meter on any graphic type page other than map (line, bar, compare) that cannot be displayed on the map.
2. Go to the map graphic page.
3. Select a meter that can be displayed.
4. Go back to another graphic type page and only the meter that was shown on the map appears and the other meter is gone.

### Note

If you select these two meters before going to the map page then they remain. The critical step appears to be selecting a meter while on the map page. Also, groups are not impacted but that may be because we currently cannot graph them on maps.

### Expected behavior
Selected meters on any graphic should remain even if they cannot be shown on another graphic.

### PR DESCRIPTION
**Pull request by spearec**
# Description

This PR changes the logic for determining when a meter/group is selected or deselected and changes the sorting for the dropdown menus. The new sorting has a divider to clearly show the user why incompatible meters are greyed out, as well as putting meters without GPS below those with GPS in the map view while preserving the original order.

image

Fixes #600 

## Type of change

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

### CODE REVIEW

File: src/client/app/components/ChartDataSelectComponent.tsx [hunk 1]
```diff
@@ -225,10 +291,24 @@ export default function ChartDataSelectComponent() {
 			<div style={divBottomPadding}>
 				<MultiSelectComponent
 					options={dataProps.sortedMeters}
-					selectedOptions={dataProps.selectedMeters}
+					selectedOptions={dataProps.displayableSelectedMeters}
 					placeholder={intl.formatMessage(messages.selectMeters)}
-					onValuesChange={(newSelectedMeterOptions: SelectOption[]) =>
-						dispatch(changeSelectedMeters(newSelectedMeterOptions.map(s => s.value)))}
+					// change logic for selection: compute diff between dataProps.selectedMeters and newSelectedMeterOptions\
```
Comment (huss) [inline src/client/app/components/ChartDataSelectComponent.tsx]: Remove \ at end of comment?

File: src/client/app/components/ChartDataSelectComponent.tsx [hunk 2]
```diff
@@ -212,10 +266,22 @@ export default function ChartDataSelectComponent() {
 			<div style={divBottomPadding}>
 				<MultiSelectComponent
 					options={dataProps.sortedGroups}
-					selectedOptions={dataProps.selectedGroups}
+					selectedOptions={dataProps.displayableSelectedGroups}
 					placeholder={intl.formatMessage(messages.selectGroups)}
-					onValuesChange={(newSelectedGroupOptions: SelectOption[]) =>
-						dispatch(changeSelectedGroups(newSelectedGroupOptions.map(s => s.value)))}
+					onValuesChange={(newSelectedGroupOptions: SelectOption[]) => {
+						const allSelectedGroupIDs: number[] = dataProps.allSelectedGroups.map(s => s.value);
```
Comment (huss) [inline src/client/app/components/ChartDataSelectComponent.tsx:277]: There are more comments in the section below but even with these I think some more would help future developers. Could you describe why each of the filters work and any other choices?

**Review by huss — CHANGES_REQUESTED** [review]
Thanks to @spearec for more good work. I like the separators on the menus and the consistent use. I have a couple of comments on commenting but nothing else. Testing indicates it works as expected.

**Comment by spearec** [comment]
I've added some more comments and changed variable names to be consistent with term usage. This makes the distinction between displayable items (an admin-level setting) and compatible items (determined by graph type/unit selection).

**Review by huss — APPROVED** [review]

**Comment by huss** [comment]
Everything looks good. Thanks.

**Pull request merged**